File: ftp/ringo/read/ftp_read.js
#! /usr/bin/ringo
// ------------------------------------------------------------------
//
//	ftp_read.js
//
//						Jun/15/2011
//
// ------------------------------------------------------------------
var text_manipulate = require ('/var/www/data_base/common/ringo_common/text_manipulate.js');
// ------------------------------------------------------------------

print	("*** 開始 ***");
var args = require('system').args;
var file_in = args[1];

var client= require ("ringo/httpclient");

client.username = 'scott';
client.password = 'tiger';
print	("*** ppp ***");

// var url='ftp://scott:tiger@host_dbase/city/iwate.json';
// var url='ftp://cdbd026/city/iwate.json';
var url='ftp://host_dbase/home/scott/city/iwate.json';

print	("*** qqq ***");
var data=client.get (url);
print	("*** rrrr ***");

print (data.status);
print (data.content);

var dict_aa = JSON.parse (data.content);

text_manipulate.dict_display_proc (dict_aa);

print	("*** 終了 ***");
// ------------------------------------------------------------------

File: ftp/cplus/read/ftp_read.cpp
// --------------------------------------------------------------------
/*
	ftp_read.cpp

					Mar/04/2014

*/
// --------------------------------------------------------------------
#include	<string>
#include	<iostream>
#include	<map>

using namespace std;
typedef map<string,string> Unit;

extern string url_get_proc (const char url[]);
extern map <string,Unit> json_to_dict_proc (string str_json);
extern	void dict_display_proc (map <string,Unit> dict_aa);
// --------------------------------------------------------------------
int main (int argc,char *argv[])
{
	cerr << "*** 開始 ***\n";

	char url_target[] = "ftp://scott:tiger@host_dbase/city/iwate.json";

	string str_json = url_get_proc (url_target);

	map <string,Unit> dict_aa = json_to_dict_proc (str_json);

	dict_display_proc (dict_aa);

	cerr << "*** 終了 ***\n";

	return 0;
}

// --------------------------------------------------------------------
